api/v1alpha1/gateway_types.go

```diff
@@ -225,29 +225,31 @@ type LocalObjectReference struct {
 	//
 	// +kubebuilder:validation:Required
 	// +required
-	Group string `json:"group"`
+	Group string `json:"group" protobuf:"bytes,1,opt,name=group"`
 	// Resource is the resource of the referent.
 	//
 	// +kubebuilder:validation:Required
 	// +required
-	Resource string `json:"resource"`
+	Resource string `json:"resource" protobuf:"bytes,2,opt,name=resource"`
 	// Name is the name of the referent.
 	//
 	// +kubebuilder:validation:Required
 	// +required
-	Name string `json:"name"`
+	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
 }
 
 // CertificateObjectReference identifies a certificate object within a known
 // namespace.
 //
 // +k8s:deepcopy-gen=false
+// +protobuf=false
 type CertificateObjectReference = LocalObjectReference
 
 // ListenerExtensionObjectReference identifies a listener extension object
 // within a known namespace.
 //
 // +k8s:deepcopy-gen=false
+// +protobuf=false
 type ListenerExtensionObjectReference = LocalObjectReference
 
 // RouteObjectReference identifies a route object within a known namespace.
```

Comment:
Should this have `+protobuf=false` too?  I'm trying to rebase #111 and getting failures:
```
api/v1alpha1/generated.pb.go:1684:33: m.Routes[iNdEx].MarshalToSizedBuffer undefined (type RouteObjectReference has no field or method MarshalToSizedBuffer)
api/v1alpha1/generated.pb.go:3155:9: e.Size undefined (type RouteObjectReference has no field or method Size)
api/v1alpha1/generated.pb.go:3758:63: f.String undefined (type RouteObjectReference has no field or method String)
api/v1alpha1/generated.pb.go:5705:21: cannot use LocalObjectReference literal (type LocalObjectReference) as type RouteObjectReference in append
api/v1alpha1/generated.pb.go:5706:39: m.Routes[len(m.Routes) - 1].Unmarshal undefined (type RouteObjectReference has no field or method Unmarshal)
```

Reply:
My understanding is that Go associates methods with the concrete types, not the alias. So we cannot have `ListenerExtensionObjectReference.DeepCopy` and `RouteObjectReference.DeepCopy`, since then there are two DeepCopy associated with `LocalObjectReference`. The go-to-protobuf generator doesn't seem to handle this well. `RouteObjectReference` is somehow deemed as the chosen alias that represents `LocalObjectReference` (possibly its the last one processed?), so if we mark it `protobuf=false` there will be no type for `LocalObjectReference` generated.

Since you are making `RouteObjectReference` no longer a type, it should have protobuf generated, but now there is probably nothing generating `LocalObjectReference`? So maybe we need to remove protobuf=false from one of these type aliases now?

Let me know if that makes sense, I can try to get things working on your PR if not

Reply:
That makes sense.  Thanks!